### PR TITLE
JAMES-2813 Add type to additional informations dto

### PR DESCRIPTION
**Pull request by remk**
Preliminary work for the serialization of the Termination events including the additional information

### CODE REVIEW

**Comment by remk** [comment]
```
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Running org.apache.james.jmap.cassandra.filtering.CassandraEventSourcingFilteringManagementTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 23.422 s - in org.apache.james.jmap.cassandra.filtering.CassandraEventSourcingFilteringManagementTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Running org.apache.james.jmap.cassandra.filtering.DTOTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.072 s - in org.apache.james.jmap.cassandra.filtering.DTOTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Running org.apache.james.jmap.cassandra.access.CassandraAccessTokenRepositoryTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [ERROR] Tests run: 9, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.171 s <<< FAILURE! - in org.apache.james.jmap.cassandra.access.CassandraAccessTokenRepositoryTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [ERROR] validTokenMustBeRetrieved  Time elapsed: 0.22 s  <<< ERROR!
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] org.apache.james.jmap.api.access.exceptions.InvalidAccessToken: 33a4a10b-a8cc-4c28-b8d8-501dcbea278f is missing
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] 
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Running org.apache.james.jmap.cassandra.vacation.CassandraNotificationRegistryTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.686 s - in org.apache.james.jmap.cassandra.vacation.CassandraNotificationRegistryTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Running org.apache.james.jmap.cassandra.vacation.CassandraVacationRepositoryTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.601 s - in org.apache.james.jmap.cassandra.vacation.CassandraVacationRepositoryTest
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] 
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] Results:
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [INFO] 
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [ERROR] Errors: 
[33eecbe744245eaadc2cde0e5dfae2b7df6aedd1] [ERROR]   CassandraAccessTokenRepositoryTest>AccessTokenRepositoryTest.validTokenMustBeRetrieved:45 » InvalidAccessToken
```
test this please

**Comment by remk** [comment]
```
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	at java.base/java.lang.Thread.run(Thread.java:834)
[7c41947a8ff092ae2a0b4ce867811f1654054048] 09:51:07.526 [ERROR] o.a.j.m.e.d.InVmEventDelivery - Error while processing listener
[7c41947a8ff092ae2a0b4ce867811f1654054048] org.apache.james.mailbox.exception.MailboxRoleNotFoundException: Could not find any mailbox with role 'spam'
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	at org.apache.james.mailbox.SystemMailboxesProvider.lambda$findMailbox$0(SystemMailboxesProvider.java:33)
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	at org.apache.james.mailbox.SystemMailboxesProvider.findMailbox(SystemMailboxesProvider.java:33)
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	at org.apache.james.mailbox.spamassassin.SpamAssassinListener.isMessageMovedToSpamMailbox(SpamAssassinListener.java:159)
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	at org.apache.james.mailbox.spamassassin.SpamAssassinListener.handleMessageMove(SpamAssassinListener.java:118)
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	at org.apache.james.mailbox.spamassassin.SpamAssassinListener.event(SpamAssassinListener.java:95)
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	at org.apache.james.mailbox.events.delivery.InVmEventDelivery.doDeliverToListener(InVmEventDelivery.java:88)
[7c41947a8ff092ae2a0b4ce867811f1654054048] 	... 11 common frames omitted
```
test this please

**Comment by rouazana** [comment]
```
[7c41947a8ff092ae2a0b4ce867811f1654054048] Error response from daemon: No such container: keystore
```
test this please

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/migration/MigrationTaskSerializationTest.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
     private static final int SCHEMA_VERSION = 12;
     private static final String SERIALIZED_TASK = "{\"type\": \"CassandraMigration\", \"targetVersion\": 12}";
-    private static final String SERIALIZED_ADDITIONAL_INFORMATION = "{\"targetVersion\": 12}";
+    private static final String SERIALIZED_ADDITIONAL_INFORMATION = "{\"type\": \"CassandraMigration\", \"targetVersion\": 12}";
```
Comment (mbaechler) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/migration/MigrationTaskSerializationTest.java]: as far as I know we decided to use kebab case for serialized types, could you fix that for all AdditionalInformation?
It could very well be done in another PR.

Comment (rouazana) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/migration/MigrationTaskSerializationTest.java]: please open a specific ticket for normalizing the types please

**Comment by rouazana** [comment]
merged

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/migration/MigrationTaskSerializationTest.java (hunk 1 above)
Comment (mbaechler) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/migration/MigrationTaskSerializationTest.java]: https://ci.linagora.com/linagora/lgs/openpaas/james/issues/1481
